src/llmcompressor/transformers/finetune/text_generation.py

```diff
@@ -111,53 +127,66 @@ def compress(**kwargs):
+        * data_args in
+            src/llmcompressor/transformers/utils/arg_parser/data_args.py
+        * recipe_args in
+            src/llmcompressor/transformers/utils/arg_parser/recipe_args.py
+        * training_args in
+            src/llmcompressor/transformers/utils/arg_parser/training_args.py
+
+    Throws deprecation warnings
+
+    :param include_training_args: Add training_args in the output if set to True.
+        Note that instantiatng trainng_args will reset HF accelerator and change its
+        internal state. This dataclass should be instantiated only once to avoid
+        conflict with Accelerate library's accelerator.
 
-    Throws depreciation warnings
     """
-    parser = HfArgumentParser(
-        (ModelArguments, DataTrainingArguments, TrainingArguments)
-    )
+    output_dir = kwargs.pop("output_dir", None)
+    parser = HfArgumentParser(_get_dataclass_arguments(include_training_args))
+
+    # parse from kwargs or cli
     if not kwargs:
-        model_args, data_args, training_args = parser.parse_args_into_dataclasses()
+        output_dir = _get_output_dir_from_argv()
+        parsed_args = parser.parse_args_into_dataclasses()
     else:
-        model_args, data_args, training_args = parser.parse_dict(kwargs)
+        parsed_args = parser.parse_dict(kwargs)
 
-    if training_args.recipe_args is not None:
-        if not isinstance(training_args.recipe_args, dict):
-            arg_dict = {}
-            for recipe_arg in training_args.recipe_args:
-                key, value = recipe_arg.split("=")
-                arg_dict[key] = value
-            training_args.recipe_args = arg_dict
+    # Unpack parsed arguments. Oneshot does not need training_args
+    if include_training_args:
+        model_args, data_args, recipe_args, training_args = parsed_args
+        if output_dir:
+            training_args.output_dir = output_dir
+    else:
+        model_args, data_args, recipe_args = parsed_args
+        training_args = None
+
+    # populate recipe arguments
+    if recipe_args.recipe_args:
+        recipe_args.recipe_args = _unwrap_recipe_args(recipe_args.recipe_args)
 
-    # raise depreciation warnings
     if data_args.remove_columns is not None:
         warnings.warn(
-            "`remove_columns` argument is depreciated. When tokenizing datasets, all "
-            "columns which are invalid inputs the tokenizer will be removed",
+            (
+                "`remove_columns` is deprecated."
+                "Invalid columns for tokenizers will be removed.",
+            ),
             DeprecationWarning,
         )
 
-    # silently assign tokenizer to processor
-    if model_args.tokenizer:
-        if model_args.processor:
-            raise ValueError("Cannot use both a tokenizer and processor")
-        model_args.processor = model_args.tokenizer
-    model_args.tokenizer = None
-
-    return model_args, data_args, training_args
+    _validate_model_args_tokenizer(model_args)
```

Comment:
This pattern of validating arguments as soon as they're received will be a common one. Can we rename this to `validate_arguments` and have it handle more than just model_args?